File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_cpp/GomokuGame.cpp
#include <iostream>
#include <vector>
#include <string>

class GomokuGame {
private:
    int boardSize;
    std::vector<std::vector<char>> board;
    char currentPlayer;

    bool _check_five_in_a_row(int row, int col, const std::vector<int>& direction) {
        int dx = direction[0];
        int dy = direction[1];
        int count = 1;
        char symbol = board[row][col];
        for (int i = 1; i < 5; i++) {
            int newRow = row + dx * i;
            int newCol = col + dy * i;
            if (newRow < 0 || newRow >= boardSize || newCol < 0 || newCol >= boardSize) {
                break;
            }
            if (board[newRow][newCol] != symbol) {
                break;
            }
            count++;
        }
        for (int i = 1; i < 5; i++) {
            int newRow = row - dx * i;
            int newCol = col - dy * i;
            if (newRow < 0 || newRow >= boardSize || newCol < 0 || newCol >= boardSize) {
                break;
            }
            if (board[newRow][newCol] != symbol) {
                break;
            }
            count++;
        }
        return count >= 5;
    }

public:
    GomokuGame(int boardSize) : boardSize(boardSize), currentPlayer('X') {
        board.resize(boardSize, std::vector<char>(boardSize, ' '));
    }

    bool make_move(int row, int col) {
        if (board[row][col] == ' ') {
            board[row][col] = currentPlayer;
            currentPlayer = (currentPlayer == 'X') ? 'O' : 'X';
            return true;
        }
        return false;
    }

    std::string checkWinner() {
        const std::vector<std::vector<int>> directions = {{0, 1}, {1, 0}, {1, 1}, {1, -1}};
        for (int row = 0; row < boardSize; row++) {
            for (int col = 0; col < boardSize; col++) {
                if (board[row][col] != ' ') {
                    for (const auto& direction : directions) {
                        if (checkFiveInARow(row, col, direction)) {
                            return std::string(1, board[row][col]);
                        }
                    }
                }
            }
        }
        return "null";
    }
};

int main() {
    GomokuGame gomokuGame(10);
    std::vector<std::vector<int>> moves = {{5,5}, {0,0}, {5,4}, {0,1}, {5,3}, {0,2}, {5,2}, {0,3}, {5,1}};
    for (const auto& move : moves) {
        gomokuGame.make_move(move[0], move[1]);
    }
    std::cout << gomokuGame.checkWinner() << std::endl;
    return 0;
}

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_py/code_output/GomokuGame.py
class GomokuGame:
    def __init__(self, board_size):
        self.board_size = board_size
        self.board = [[' ' for _ in range(board_size)] for _ in range(board_size)]
        self.current_player = 'X'

    def make_move(self, row, col):
        if self.board[row][col] == ' ':
            self.board[row][col] = self.current_player
            self.current_player = 'O' if self.current_player == 'X' else 'X'
            return True
        return False

    def check_winner(self):
        directions = [(0, 1), (1, 0), (1, 1), (1, -1)]
        for row in range(self.board_size):
            for col in range(self.board_size):
                if self.board[row][col] != ' ':
                    for direction in directions:
                        if self.check_five_in_a_row(row, col, direction):
                            return self.board[row][col]
        return None

    def check_five_in_a_row(self, row, col, direction):
        dx, dy = direction
        count = 1
        symbol = self.board[row][col]
        
        for i in range(1, 5):
            new_row = row + dx * i
            new_col = col + dy * i
            if not (0 <= new_row < self.board_size and 0 <= new_col < self.board_size):
                break
            if self.board[new_row][new_col] != symbol:
                break
            count += 1
            
        for i in range(1, 5):
            new_row = row - dx * i
            new_col = col - dy * i
            if not (0 <= new_row < self.board_size and 0 <= new_col < self.board_size):
                break
            if self.board[new_row][new_col] != symbol:
                break
            count += 1
            
        return count >= 5

if __name__ == '__main__':
    gomoku_game = GomokuGame(10)
    moves = [[5, 5], [0, 0], [5, 4], [0, 1], [5, 3], [0, 2], [5, 2], [0, 3], [5, 1]]
    for move in moves:
        gomoku_game.make_move(move[0], move[1])
    print(gomoku_game.check_winner())